sdk/keyvault/azure-security-keyvault-keys/migration_guide.md

```diff
@@ -0,0 +1,226 @@
+
+In the case of the Key Vault, the modern client libraries have packages and namespaces that begin with `com.azure.security.keyvault` and were released beginning with version `4.0.0`. The legacy client libraries have packages and namespaces that begin with `com.microsoft.azure.keyvault` and a version of `1.x.x` or below.
+
+### Client constructors
+Across all modern Azure client libraries, clients consistently take an endpoint or connection string along with token credentials. This differs from `KeyVaultClient`, which took an authentication delegate and could be used for multiple Key Vault endpoints.
+
+### Authentication
+Previously in `azure-keyvault` you could create a `KeyVaultClient` by passing either a `KeyVaultCredential` or `RestClient` from `client-runtime`:
+
+```java
+import com.microsoft.azure.keyvault.KeyVaultClient;
+import com.microsoft.azure.keyvault.authentication.KeyVaultCredentials;
+
+KeyVaultCredentials keyVaultCredentials = new MyKeyVaultCredentials("<client-id>", "<client-key>");
+KeyVaultClient keyVaultClient = new KeyVaultClient(keyVaultCredentials);
+```
+
+Now in `azure-security-keyvault-keys` you can create a `KeyClient` using any credential from [`azure-identity`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/identity/azure-identity/README.md). Below is an example using [`DefaultAzureCredential`](https://docs.microsoft.com/en-us/java/api/overview/azure/identity-readme?view=azure-java-stable#defaultazurecredential):
+
+```java
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.security.keyvault.keys.KeyClient;
+import com.azure.security.keyvault.keys.KeyClientBuilder;
+
+TokenCredentials tokenCredentials = new DefaultAzureCredentialBuilder().build();
+
+KeyClient keyClient = new KeyClientBuilder()
+    .vaultUrl("<your-key-vault-url>")
+    .credential(tokenCredentials)
+    .buildClient();
+```
+
+### Async operations
+The modern `azure-security-keyvault-keys` library includes a complete set of async APIs that return [Project Reactor -based types](https://projectreactor.io/), as opposed to `azure-keyvault` async APIs that return either [Observable](reactivex.io/RxJava/javadoc/io/reactivex/Observable.html) or [ServiceFuture](http://azure.github.io/ref-docs/java/com/microsoft/rest/ServiceFuture.html).
+
+Another difference is that async operations are available on their own separate async clients, which include the word `Async` in their name, like `KeyAsyncClient`.
+
+### Common scenarios
+Repeat this section for the common high level usage scenarios for this library.
+Show how you would accomplish these both in the old and new packages, pointing out the key differences, reasons and advantages.
```

Comment:
```suggestion
```

Reply:
@ramya-rao-a Do you think we should have everything that's currently under "Common scenarios" be under "Important changes", at the same level as everything else there?

Reply:
I'd like to keep the header at least.